pandas/core/indexes/base.py

```diff
@@ -1526,6 +1527,7 @@ def _set_names(self, values, level=None) -> None:
 
     names = property(fset=_set_names, fget=_get_names)
 
+    @deprecate_nonkeyword_arguments(version="2.0", allowed_args=["self", "names"])
```

Comment:
Can you put `version=None`? I know I'd originally put `version="2.0"`, but then other reviewers expressed a preference for not specifying the exact version.

Reply:
Added in my latest commit.

I think these changes also apply to the PR I opened for drop_duplicates (#41500)

Would you like me to make these changes there too?

Reply:
Yes please, thanks! Your help with these is much appreciated